### ISSUE TITLE
exporting certificate chain produces empty files

### ISSUE THREAD

**Issue by henningkessler** [chris2511/xca#309]
Hi,

when I export a certificate and select "PEM Kette (*.pem)" an empty file is created. This happens under macOS 11.5.2 with xca 2.4.0.

Henning

**Comment by footstep86** [comment]
Same here: XCA 2.4.0 running on macOS 11.6.

**Comment by zaphod80013** [comment]
Also seeing this on MacOS 10.15.7 I was able to work around this by using the Export "PEM All" option. 

Well all I can say for certain is that after exporting with 'PEM All' I was able to import a root ca along with tier-1 & tier-2 intermediate certificate authorities into the OSX Keychain and they look correct.

**Comment by nomagick** [comment]
Same issue here
Version: 2.4.0

**Comment by SemoTech** [comment]
Confirmed version 2.4.0 of XCA has an export bug for "PEM chain (*.pem)" that creates 0-byte empty files! 
Tested on MacOS Monterey 12.0.1

Temporary workaround until this is fixed in the code:

If you need a .pem chain export, export 1st using "PKCS #12 (pfx)" with a password, then issue the command below and re-enter the same password:

`openssl pkcs12 -in your-cert-name.pfx -out your-cert-name.pem `

**Comment by wh0else** [comment]
Any updates on this?

**Comment by SemoTech** [comment]
Feels like XCA is abandoned and thats a big shame...

**Comment by wh0else** [comment]
Bummer! Liked it alot. Anyone have any good alternatives?

**Comment by SemoTech** [comment]
The feeling is mutual.  Very disappointed.  If someone has anything similar (ideally for Mac), please share....

**Comment by meermanr** [comment]
FWIW, the most recent (unstable) build seems to fix this for me.

I found it from this discussion thread: [Is a new version coming soon?... · chris2511/xca · Discussion #432](https://github.com/chris2511/xca/discussions/432)

This took me to [Fixup for a548227a073: install translation files again. · chris2511/xca@8983e50](https://github.com/chris2511/xca/actions/runs/5088625840) from which I downloaded the [macos-6.5.1](https://github.com/chris2511/xca/suites/13166803636/artifacts/716318326) artifact and installed it. It didn't "just work", to get it running I had to do two things:

# Remove the Apple quarantine warnings
# Launch it from a terminal with `--database=`

I installed it as `/Applications/xca-beta`, so the command to remove all quarantine markers was `find /Applications/xca-beta.app/ -print0 | xargs -r0n1 xattr -d -r com.apple.quarantine`.

Launching from the terminal was done as:

```sh
cd /Applications/xca-beta.app/Contents/MacOS
./xca '--database=~/certificates_xca.xdb'
```

Without `--database` it seemed to immediately close without displaying the GUI.

**Comment by SemoTech** [comment]
Good news! I believe this Chain Export issue is finally fixed in the new XCA v2.5.0 that was released today!!!

**Comment by wh0else** [comment]
Fixed in v2.5.0
